consensus/state.go

```diff
@@ -1213,14 +1217,18 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 	// If a block is locked, prevote that.
 	if cs.LockedBlock != nil {
 		logger.Info("enterPrevote: Already locked on a block, prevoting locked block")
-		cs.signAddVote(tmproto.PrevoteType, cs.LockedBlock.Hash(), cs.LockedBlockParts.Header())
+		cs.signAddVote(tmproto.PrevoteType, types.BlockID{
+			Hash:                   cs.LockedBlock.Hash(),
+			PartSetHeader:          cs.LockedBlockParts.Header(),
+			DataAvailabilityHeader: &cs.LockedBlock.DataAvailabilityHeader},
+		)
 		return
 	}
 
 	// If ProposalBlock is nil, prevote nil.
 	if cs.ProposalBlock == nil {
 		logger.Info("enterPrevote: ProposalBlock is nil")
-		cs.signAddVote(tmproto.PrevoteType, nil, types.PartSetHeader{})
+		cs.signAddVote(tmproto.PrevoteType, types.EmptyBlockID())
```

Comment:
That's a good point. Personally, I think `EmptyBlockID` makes sense because its the block id for a block with no data, not a block id for a nil block. Definitely not attached to that name by any means, though.

Reply:
Yeah, from this perspective it also makes sense, so here we have a semantical meaning tradeoff or something. I am ok with any option btw)